homework/index.js

```diff
@@ -29,22 +29,74 @@
     return elem;
   }
 
-  function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+  function createTag(name, options = {}) {
+    const elem = document.createElement(name);
+    Object.entries(options).forEach(([key, value]) => {
+      if (key === 'text') {
+        elem.textContent = value;
+      } else {
+        elem.setAttribute(key, value);
+      }
+    });
+    return elem;
+  }
+
+  function buildTableRow(title, parent, content) {
+    const row = createAndAppend(`tr`, parent);
+    createAndAppend(`th`, row, { text: `${title}:` });
+    let data;
+    if (typeof content === `string` || typeof content === `number`) {
+      createAndAppend(`td`, row, { text: `${content}` });
+    } else {
+      data = createAndAppend(`td`, row);
+      data.appendChild(content);
+    }
+  }
+
+  function renderRepoDetails(repo, parent) {
+    const detailsDiv = createAndAppend(`div`, parent, { class: `details` });
+    const table = createAndAppend(`table`, detailsDiv);
+
+    const repoLink = createTag(`a`, {
+      href: repo.html_url,
+      target: '_blank',
+      text: repo.name,
+    });
+    buildTableRow(`Repository`, table, repoLink);
```

Comment:
Instead of passing a pre-made `<a>` tag to the `buildTableRow` function, you can add it afterwards (using the simplified `buildTableRow` function), like this:

```js
const firstRow = buildTableRow(`Repository`, table, '');
createAndAppend(`a`, firstRow.lastChild, {
  href: repo.html_url,
  target: '_blank',
  text: repo.name,
});
```

Reply:
Wow! I didn't think about this.